bcgov/cas-registration
# Fix/Eslint errors after monorepo immigration

Tying up loose linting ends from bcgov/cas-obps#257. Nx linting rules were inconsistent with our existing AirBNB-TypeScript + NextJS rules, and moving of files triggered some other linting issues. 

## Changes 🚧

- Removed all `@NX/` linting rules
  - _Except_ `@nx/enforce-module-boundries`, as it is specific to monorepo structuring
- Updated `eslintrc` and `tsconfig` configs in Apps and Libs to be more consistent with pre-monorepo usage
- Addressed linting errors that were still being flagged

## To test 🔬

> You may need to run `yarn` in the `./bciers` directory to install and update packages.
- Navigate to `./bciers` and run (if nx is not installed globally, first `npx `)`nx run reporting:lint`. All files should pass linting. 
- _Optional_ Type in a line you know will cause linting errors in a file, (if configured) your IDE should flag the linting issues. ie. `var failure: number = "string";

## Notes 📝

Starting with Eslint 9.0.0, it is being changed to use a [flat config file](https://eslint.org/docs/latest/use/configure/migration-guide#start-using-flat-config-files) rather than the existing format. Though 8.57.0 already supports this format, NextJS 14 still has compatability issues. Once NextJS adds full support, we should also make the move. There is a [Nx Generator](https://nx.dev/recipes/tips-n-tricks/flat-config) that can do much of the lift for us. Flagged this in a [tech-debt card](https://github.com/orgs/bcgov/projects/123/views/1?pane=issue&itemId=62005287).
